src/Rules/Properties/AccessPropertiesRule.php

```diff
@@ -41,101 +43,104 @@ public function getNodeType(): string
+			if ($type instanceof ErrorType) {
+				$errors = array_merge($errors, $typeResult->getUnknownClassErrors());
+				continue;
 			}
-		);
-		$type = $typeResult->getType();
-		if ($type instanceof ErrorType) {
-			return $typeResult->getUnknownClassErrors();
-		}
 
-		if ($scope->isInExpressionAssign($node)) {
-			return [];
-		}
+			if ($scope->isInExpressionAssign($node)) {
+				continue;
+			}
 
-		if (!$type->canAccessProperties()->yes()) {
-			return [
-				RuleErrorBuilder::message(sprintf(
+			if (!$type->canAccessProperties()->yes()) {
+				$errors[] = RuleErrorBuilder::message(sprintf(
 					'Cannot access property $%s on %s.',
 					$name,
 					$type->describe(VerbosityLevel::typeOnly())
-				))->build(),
-			];
-		}
-
-		if (!$type->hasProperty($name)->yes()) {
-			if ($scope->isSpecified($node)) {
-				return [];
+				))->build();
+				continue;
 			}
 
-			$classNames = $typeResult->getReferencedClasses();
-			if (!$this->reportMagicProperties) {
-				foreach ($classNames as $className) {
-					if (!$this->reflectionProvider->hasClass($className)) {
-						continue;
-					}
+			if (!$type->hasProperty($name)->yes()) {
+				if ($scope->isSpecified($node)) {
+					continue;
+				}
 
-					$classReflection = $this->reflectionProvider->getClass($className);
-					if (
-						$classReflection->hasNativeMethod('__get')
-						|| $classReflection->hasNativeMethod('__set')
-					) {
-						return [];
+				$classNames = $typeResult->getReferencedClasses();
+				if (!$this->reportMagicProperties) {
+					foreach ($classNames as $className) {
+						if (!$this->reflectionProvider->hasClass($className)) {
+							continue;
+						}
+
+						$classReflection = $this->reflectionProvider->getClass($className);
+						if (
+							$classReflection->hasNativeMethod('__get')
+							|| $classReflection->hasNativeMethod('__set')
+						) {
+							continue 2;
+						}
 					}
 				}
-			}
 
-			if (count($classNames) === 1) {
-				$referencedClass = $typeResult->getReferencedClasses()[0];
-				$propertyClassReflection = $this->reflectionProvider->getClass($referencedClass);
-				$parentClassReflection = $propertyClassReflection->getParentClass();
-				while ($parentClassReflection !== false) {
-					if ($parentClassReflection->hasProperty($name)) {
-						return [
-							RuleErrorBuilder::message(sprintf(
+				if (count($classNames) === 1) {
+					$referencedClass = $typeResult->getReferencedClasses()[0];
+					$propertyClassReflection = $this->reflectionProvider->getClass($referencedClass);
+					$parentClassReflection = $propertyClassReflection->getParentClass();
+					while ($parentClassReflection !== false) {
+						if ($parentClassReflection->hasProperty($name)) {
+							$errors[] = RuleErrorBuilder::message(sprintf(
 								'Access to private property $%s of parent class %s.',
 								$name,
 								$parentClassReflection->getDisplayName()
-							))->build(),
-						];
-					}
+							))->build();
+							continue 2;
```

Comment:
dtto

Reply:
Here we really need "continue 2" otherwise ```while ($parentClassReflection !== false) {``` can't be escaped anymore. ;)